src/runtime_src/core/tools/xbmgmt2/SubCmdFlash.cpp

```diff
@@ -163,17 +233,159 @@ bool canProceed()
   return proceed;
 }
 
+/* 
+ * Flash shell and sc firmware
+ * Helper method for auto_flash
+ */
+static int 
+updateShellAndSC(uint16_t boardIdx, DSAInfo& candidate, bool& reboot)
+{
+  reboot = false;
+
+  Flasher flasher(boardIdx);
+  if(!flasher.isValid()) {
+    std::cout << "card not available" << std::endl;
+    return -EINVAL;
+  }
+
+  bool same_dsa = false;
+  bool same_bmc = false;
+  DSAInfo current = flasher.getOnBoardDSA();
+  if (!current.name.empty()) {
+    same_dsa = (candidate.name == current.name &&
+      candidate.matchId(current));
+    same_bmc = (current.bmcVer.empty() ||
+      candidate.bmcVer == current.bmcVer);
+  }
+  if (same_dsa && same_bmc)
+    std::cout << "update not needed" << std::endl;
+
+  if (!same_bmc) {
+    std::cout << "Updating SC firmware on card[" << flasher.sGetDBDF() <<
+      "]" << std::endl;
+    auto ret = 0;update_SC(boardIdx, candidate.file.c_str());
+    if (ret != 0) {
+      std::cout << "WARNING: Failed to update SC firmware on card ["

```

Comment:
Hmm..It feels like if we failed one operation we shouldn't be continuing the flow.  Also it should be an error not warning.

Reply:
I'm not sure about breaking the flow here. Sometimes SC fails to update but the shell does and the card works fine.